src/backend/distributed/deparser/deparse_foreign_server_stmts.c

```diff
@@ -0,0 +1,304 @@
+static void AppendAlterForeignServerStmt(StringInfo buf, AlterForeignServerStmt *stmt);
+static void AppendAlterForeignServerOptions(StringInfo buf, AlterForeignServerStmt *stmt);
+static void AppendAlterForeignServerRenameStmt(StringInfo buf, RenameStmt *stmt);
+static void AppendAlterForeignServerOwnerStmt(StringInfo buf, AlterOwnerStmt *stmt);
+static void AppendDropForeignServerStmt(StringInfo buf, DropStmt *stmt);
+static void AppendServerNames(StringInfo buf, DropStmt *stmt);
+static void AppendBehavior(StringInfo buf, DropStmt *stmt);
+static char * GetDefElemActionString(DefElemAction action);
+
+char *
+DeparseCreateForeignServerStmt(Node *node)
+{
+	CreateForeignServerStmt *stmt = castNode(CreateForeignServerStmt, node);
+
+	StringInfoData str;
+	initStringInfo(&str);
+
+	AppendCreateForeignServerStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+char *
+DeparseAlterForeignServerStmt(Node *node)
+{
+	AlterForeignServerStmt *stmt = castNode(AlterForeignServerStmt, node);
+
+	StringInfoData str;
+	initStringInfo(&str);
+
+	AppendAlterForeignServerStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+char *
+DeparseAlterForeignServerRenameStmt(Node *node)
+{
+	RenameStmt *stmt = castNode(RenameStmt, node);
+
+	Assert(stmt->renameType == OBJECT_FOREIGN_SERVER);
+
+	StringInfoData str;
+	initStringInfo(&str);
+
+	AppendAlterForeignServerRenameStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+char *
+DeparseAlterForeignServerOwnerStmt(Node *node)
+{
+	AlterOwnerStmt *stmt = castNode(AlterOwnerStmt, node);
+
+	Assert(stmt->objectType == OBJECT_FOREIGN_SERVER);
+
+	StringInfoData str;
+	initStringInfo(&str);
+
+	AppendAlterForeignServerOwnerStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+char *
+DeparseDropForeignServerStmt(Node *node)
+{
+	DropStmt *stmt = castNode(DropStmt, node);
+
+	Assert(stmt->removeType == OBJECT_FOREIGN_SERVER);
+
+	StringInfoData str;
+	initStringInfo(&str);
+
+	AppendDropForeignServerStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+static void
+AppendCreateForeignServerStmt(StringInfo buf, CreateForeignServerStmt *stmt)
+{
+	appendStringInfoString(buf, "CREATE SERVER ");
+
+	if (stmt->if_not_exists)
+	{
+		appendStringInfoString(buf, "IF NOT EXISTS ");
+	}
+
+	appendStringInfo(buf, "%s ", stmt->servername);
+
+	if (stmt->servertype)
+	{
+		appendStringInfo(buf, "TYPE %s ", quote_literal_cstr(stmt->servertype));
+	}
+
+	if (stmt->version)
+	{
+		appendStringInfo(buf, "VERSION %s ", quote_literal_cstr(stmt->version));
+	}
+
+	appendStringInfo(buf, "FOREIGN DATA WRAPPER %s ", stmt->fdwname);
+
+	AppendCreateForeignServerOptions(buf, stmt);
+}
+
+
+static void
+AppendCreateForeignServerOptions(StringInfo buf, CreateForeignServerStmt *stmt)
+{
+	if (list_length(stmt->options) <= 0)
+	{
+		return;
+	}
+
+	appendStringInfoString(buf, "OPTIONS (");
+
+	DefElem *def = NULL;
+	foreach_ptr(def, stmt->options)
+	{
+		const char *value = defGetString(def);
+		appendStringInfo(buf, "%s \'%s\'", def->defname, value);
```

Comment:
similarly: should we call `quote_literal_cstr` for name/value?

Reply:
Defnames don't have quotes according to pg syntax.
Values have single quotes. I've added quote_identifier for that